diff --git a/configure b/configure
index 8569a60bf8..928b19db69 100644
--- a/configure
+++ b/configure
@@ -5192,3 +5192,3 @@ probe_cc(){
         else
-            _DEPCMD='$(DEP$(1)) $(DEP$(1)FLAGS) $($(1)DEP_FLAGS) $< 2>&1 | awk '\''/including/ { sub(/^.*file: */, ""); gsub(/\\/, "/"); if (!match($$0, / /)) print "$@:", $$0 }'\'' > $(@:.o=.d)'
+            _DEPCMD='$(DEP$(1)) $(DEP$(1)FLAGS) $($(1)DEP_FLAGS) $< 2>&1 | awk -f ./msvc_dep.awk > $(@:.o=.d)'
         fi
diff --git a/msvc_dep.awk b/msvc_dep.awk
new file mode 100644
index 0000000000..a791efe000
--- /dev/null
+++ b/msvc_dep.awk
@@ -0,0 +1 @@
+/including/ { sub(/^.*file: */, ""); gsub(/\\/, "/"); if (!match($$0, / /)) print "$@:", $$0 }
